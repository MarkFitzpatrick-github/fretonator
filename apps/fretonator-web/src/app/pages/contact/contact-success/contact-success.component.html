<div class="success__container" #scrollTarget>
  <h1 class="success__title">You are amazing! <span role="img" aria-label="raising hands emoji">🙌🏼️</span></h1>
  <p class="success__copy">I can't thank you enough for your feedback. I will endeavour to reply to all messages when I
    can.</p>
  <hr class="success__divider"/>
  <h2 class="success__subtitle">Want to get involved?</h2>
  <p class="success__copy">The Fretonator is open source! If you're a developer and you'd like to contribute to the
    Fretonator,
    you can always <a href="https://github.com/ishythefishy/fretonator"
                      rel="nofollow noopener norefferer"
                      target="_blank"
                      class="success__link">fork the
      repository</a>, do your magic, and send over a
    pull request.</p>
  <a [routerLink]="['/']"
     class="success__cta"
     (click)="onClick()">Return to the Fretonator</a>
</div>
