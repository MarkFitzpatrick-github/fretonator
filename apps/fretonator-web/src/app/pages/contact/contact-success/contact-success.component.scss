@import '../../../../styles/mixins';
@import '../../../../styles/typography';

.success__container {
  max-width: $screen-med;
  margin-left: auto;
  margin-right: auto;
  background-color: var(--form-container-bg-color);
  padding: pxToRem($grid-unit * 4);

  @media screen and (min-width: $screen-med) {
    padding: pxToRem($grid-unit * 6) pxToRem($grid-unit * 16);
  }
}

.success__title {
  @include font_large;
  color: var(--form-container-color);
  text-align: left;
  margin-top: pxToRem($grid-unit * 3);
  margin-bottom: pxToRem($grid-unit * 3);
}

.success__subtitle {
  @include font_medium;
  color: var(--form-container-color);
  text-align: left;
  margin-top: pxToRem($grid-unit * 2);
  margin-bottom: pxToRem($grid-unit * 2);
}

.success__copy {
  @include font_bodyCopy;
  color: var(--form-container-color);
  text-align: left;
  margin-bottom: pxToRem($grid-unit * 2);
}

.success__link {
  @include text_link;
  color: var(--peach);
}

.success__divider {
  height: pxToRem(4);
  background-color: var(--peach);
  margin-top: pxToRem($grid-unit * 4);
  margin-bottom: pxToRem($grid-unit * 4);
}

.success__cta {
  @include hard_button_base();
  margin-top: pxToRem($grid-unit * 4);
  margin-bottom: pxToRem($grid-unit * 4);
}

