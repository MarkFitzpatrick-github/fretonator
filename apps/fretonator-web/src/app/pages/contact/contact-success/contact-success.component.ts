import { AfterViewInit, Component, ElementRef, OnInit, ViewChild } from '@angular/core';
import { GlobalService } from '../../../global.service';

@Component({
  selector: 'app-contact-success',
  templateUrl: './contact-success.component.html',
  styleUrls: ['./contact-success.component.scss']
})
export class ContactSuccessComponent implements AfterViewInit {
  @ViewChild('scrollTarget') scrollTarget: ElementRef<HTMLElement>;

  constructor(private globalService: GlobalService) {
  }

  ngAfterViewInit(): void {
    this.globalService.setScrollTarget(this.scrollTarget.nativeElement);
  }

  onClick() {
    this.globalService.getScrollTarget().scrollIntoView();
  }
}
