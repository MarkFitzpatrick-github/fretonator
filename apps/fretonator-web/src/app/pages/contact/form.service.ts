import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { retry } from 'rxjs/operators';

@Injectable(
)
export class FormService {

  constructor(private httpClient: HttpClient) {
  }

  submit(location, form) {

    const headers = new HttpHeaders()
      .set('Content-Type', 'application/x-www-form-urlencoded');

    const options = {
      headers,
      responseType: 'text' as 'text',
    };

    const formBody = new URLSearchParams();

    Object.entries(form)
      .forEach(([key, value]) =>
        formBody.append(key, value as string));

    return this.httpClient.post(location, formBody.toString(), options)
      .pipe(retry(2));
  }
}
