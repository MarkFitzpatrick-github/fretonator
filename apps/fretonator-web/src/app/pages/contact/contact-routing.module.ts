import { NgModule } from '@angular/core';
import { Route, RouterModule, Routes } from '@angular/router';
import { ContactIndexComponent } from './contact-index/contact-index.component';
import { ContactSuccessComponent } from './contact-success/contact-success.component';
import { ScrollToGuard } from '../../common/scroll-to/scroll-to.guard';

const contact: Route = {
  path: '',
  pathMatch: 'full',
  canActivate: [
    ScrollToGuard
  ],
  component: ContactIndexComponent
};

const success: Route = {
  path: 'success',
  component: ContactSuccessComponent,
  canActivate: [
    ScrollToGuard
  ]
};

const routes: Routes = [contact, success];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class ContactRoutingModule {
}
