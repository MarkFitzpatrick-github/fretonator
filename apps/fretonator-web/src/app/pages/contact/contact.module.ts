import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';

import { ContactRoutingModule } from './contact-routing.module';
import { ContactIndexComponent } from './contact-index/contact-index.component';
import { ContactSuccessComponent } from './contact-success/contact-success.component';
import { HttpClientModule } from '@angular/common/http';
import { FormService } from './form.service';


@NgModule({
  declarations: [ContactIndexComponent, ContactSuccessComponent],
  imports: [
    CommonModule,
    ContactRoutingModule,
    ReactiveFormsModule,
    FormsModule,
    HttpClientModule
  ],
  providers: [
    FormService
  ]
})
export class ContactModule { }
