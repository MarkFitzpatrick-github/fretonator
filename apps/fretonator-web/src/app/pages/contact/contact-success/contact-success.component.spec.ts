import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ContactSuccessComponent } from './contact-success.component';
import { Component } from '@angular/core';
import { ContactModule } from '../contact.module';
import { RouterTestingModule } from '@angular/router/testing';
import { CommonModule } from '@angular/common';

describe('ContactSuccessComponent', () => {
  @Component({
    selector: 'app-contact-success-spec',
    template: `
      <app-contact-success></app-contact-success>
    `
  })
  class ContactSuccessComponentSpec {
  }

  let component: ContactSuccessComponentSpec;
  let fixture: ComponentFixture<ContactSuccessComponentSpec>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ContactSuccessComponentSpec],
      imports: [
        ContactModule,
        RouterTestingModule,
        CommonModule
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ContactSuccessComponentSpec);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
