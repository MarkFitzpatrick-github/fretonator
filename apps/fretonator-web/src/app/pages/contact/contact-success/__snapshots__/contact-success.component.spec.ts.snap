// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactSuccessComponent should create 1`] = `ContactSuccessComponentSpec {}`;
